x-pack/plugins/reporting/server/core.ts

```diff
@@ -4,34 +4,42 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import Hapi from 'hapi';
 import * as Rx from 'rxjs';
 import { first, map, take } from 'rxjs/operators';
 import {
   BasePath,
   ElasticsearchServiceSetup,
+  HttpServiceSetup,
   IRouter,
   KibanaRequest,
   SavedObjectsClientContract,
   SavedObjectsServiceStart,
   UiSettingsServiceStart,
-} from 'src/core/server';
+} from '../../../../src/core/server';
 import { PluginSetupContract as FeaturesPluginSetup } from '../../features/server';
 import { LicensingPluginSetup } from '../../licensing/server';
 import { SecurityPluginSetup } from '../../security/server';
+import { SpacesPluginSetup } from '../../spaces/server';
 import { ReportingConfig } from './';
 import { HeadlessChromiumDriverFactory } from './browsers/chromium/driver_factory';
 import { checkLicense, getExportTypesRegistry } from './lib';
 import { ESQueueInstance } from './lib/create_queue';
 import { screenshotsObservableFactory, ScreenshotsObservableFn } from './lib/screenshots';
 import { ReportingStore } from './lib/store';
 
-export interface ReportingInternalSetup {
+export interface PluginSetupOpts {
   features: FeaturesPluginSetup;
   elasticsearch: ElasticsearchServiceSetup;
   licensing: LicensingPluginSetup;
-  basePath: BasePath['get'];
-  router: IRouter;
   security?: SecurityPluginSetup;
+  spaces?: SpacesPluginSetup;
+  http: HttpServiceSetup;
+}
+
+export interface ReportingInternalSetup extends Omit<PluginSetupOpts, 'http'> {
+  basePath: Pick<BasePath, 'set' | 'get'>;
+  router: IRouter;
```

Comment:
The `http` dependency is passed to our plugin setup, but the actual object that is held internally has the 2 `basePath` and `router` dependencies, instead of the single `http` dependency. This choice had a lot of impact and required a lot of supplemental changes in things like the mocks